home/components/AdditionalTwoFactorOptionsButton.tsx

```diff
@@ -0,0 +1,77 @@
+import { connectActionSheet } from '@expo/react-native-action-sheet';
+import React from 'react';
+import { TouchableOpacity, ActionSheetIOSOptions } from 'react-native';
+
+import PrimaryButton from '../components/PrimaryButton';
+
+export enum UserSecondFactorDeviceMethod {
+  AUTHENTICATOR = 'authenticator',
+  SMS = 'sms',
+}
+
+export type SecondFactorDevice = {
+  id: string;
+  method: UserSecondFactorDeviceMethod;
+  sms_phone_number: string | null;
+  is_primary: boolean;
+};
+
+type AdditionalTwoFactorOptionsButtonProps = {
+  secondFactorDevices: SecondFactorDevice[];
+  onSelectSMSSecondFactorDevice: (device: SecondFactorDevice) => void;
+  onSelectAuthenticatorOption: () => void;
+};
+
+function AdditionalTwoFactorOptionsButton({
+  secondFactorDevices,
+  onSelectSMSSecondFactorDevice,
+  onSelectAuthenticatorOption,
+  showActionSheetWithOptions,
+}: AdditionalTwoFactorOptionsButtonProps & {
+  showActionSheetWithOptions: (
+    options: ActionSheetIOSOptions,
+    callback: (buttonIndex: number) => void
+  ) => void;
+}) {
+  const handlePress = () => {
+    const hasAuthenticatorSecondFactorDevices = secondFactorDevices.find(
+      device => device.method === UserSecondFactorDeviceMethod.AUTHENTICATOR
+    );
+
+    const smsSecondFactorDevices = secondFactorDevices.filter(
+      device => device.method === UserSecondFactorDeviceMethod.SMS
+    );
+
+    const deviceOptions = smsSecondFactorDevices.map(device => device.sms_phone_number!);
+    const options = [
+      ...deviceOptions,
+      ...(hasAuthenticatorSecondFactorDevices ? ['Authenticator'] : []),
+      'Cancel',
+    ];
+    const cancelButtonIndex = options.indexOf('Cancel');
```

Comment:
```suggestion
    const cancelButtonIndex = options.lastIndexOf('Cancel');
```
just in case one day we display device names and someone names theirs Cancel